server/src/test/java/org/elasticsearch/action/search/MockSearchPhaseContext.java

```diff
@@ -111,6 +111,8 @@ public void sendSearchResponse(SearchResponseSections internalSearchResponse, At
     @Override
     public void onPhaseFailure(SearchPhase phase, String msg, Throwable cause) {
         phaseFailure.set(cause);
+        Releasables.close(releasables);
```

Comment:
Not sure if this was left intentionally out or not, but added this to be in sync with the  `AbstractAsyncSearchAction` as we do something similar through the 
```
        this.listener = ActionListener.runAfter(listener, () -> Releasables.close(releasables));

```

Reply:
@original-brownbear would know best.

Reply:
Why do we need to do this here? We do it via the listener in the async search action since we know for sure that that thing will always be resolved unless there's a bug, shouldn't it be the same here?

Reply:
IIUCT the provided `Runnable` for `runAfter` also runs on the `onFailure`. If that's the case, I was wondering whether we should also call this as part of the `onPhaseFailure` here (as the said `listener` is invoked at the end of `org.elasticsearch.action.search.AbstractSearchAsyncAction#raisePhaseFailure` when an exception is thrown) 🤔 
Don't have a strong opinion on this, happy to revert if it's not needed (for completeness).

Reply:
If I understand what you're saying correctly, doing it here would just be an optimization so it runs earlier, but in the end the listener has this covered?
If so, I'd say lets remove this.

Reply:
The actual implementation of the `onPhaseFailure` in `AbstractSearchAsyncAction` has this:
```
    private void raisePhaseFailure(SearchPhaseExecutionException exception) {
        results.getSuccessfulResults().forEach((entry) -> {
               ...
        });
        listener.onFailure(exception);
    }
```
where there `listener` closes all releasables as per the above.

In the `MockSearchPhaseContext` implementation, we have the following (i.e. setting just an atomic reference) which is why I thought that maybe it was missing:
```
    @Override
    public void onPhaseFailure(SearchPhase phase, String msg, Throwable cause) {
        phaseFailure.set(cause);
    }
```

Reply:
Ah I see :) Right now we're handling the release via `sendSearchResponse`.
This is only needed to release the ref-counted value that is seemingly redundant and that I asked about right?

Reply:
Yeap, that was a leftover, just verifying before updating :) 

My main point here was not just for this test, but overall if we were missing something. Will happily revert though, if you think it's not needed (or that it might hide some issues).